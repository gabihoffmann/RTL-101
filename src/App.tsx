function App() {
  return (
    <div>
      <h1>Hello world!</h1>
      {/* wrong text */}
    </div>
  );
}

export default App;
