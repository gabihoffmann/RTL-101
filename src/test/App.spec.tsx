import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";

import App from "../App";

describe("Test the App successfully", () => {
  it("Should render the App component", () => {
    render(<App />);
    // prints out the jsx in the App component unto the command line
    screen.debug();
  });

  it("Should show the text 'Hello world!' in the document", () => {
    render(<App />);
    const hello = screen.getByText(/hello world!/i);
    expect(hello).toBeInTheDocument();
  });

  it("Should matches the snapshot", () => {
    const { container } = render(<App />);
    // use: press u to update snapshot
    expect(container).toMatchSnapshot();
  });
});
