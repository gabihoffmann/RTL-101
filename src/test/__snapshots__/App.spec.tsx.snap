// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test the App successfully > Should matches the snapshot 1`] = `
<div>
  <div>
    <h1>
      Hello world!
    </h1>
  </div>
</div>
`;
